tools/identity-resolution/identity-resolution/Helpers/GitHubAadConverter.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Net.Http;
+using Azure.Sdk.Tools.NotificationConfiguration.Models;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Azure.Sdk.Tools.NotificationConfiguration.Helpers
+{
+    public class GitHubToAadConverter
+    {
+        /// <summary>
+        /// GitHubToAadConverter constructor for generating new token, and initialize http client.
+        /// </summary>
+        /// <param name="clientId">AAD Client/App Id</param>
+        /// <param name="clientSecret">AAD Client/App Secret</param>
+        /// <param name="tenantID">AAD Tenant Id</param>
+        /// <param name="logger">Logger</param>
+        public GitHubToAadConverter(
+            string clientId,
+            string clientSecret,
+            string tenantID,
+            ILogger<GitHubNameResolver> logger)
+        {
+            try
+            {
+                opsAuthToken = new AadTokenCredential(clientId, clientSecret, tenantID).GenerateAadToken();
+            }
+            catch (Exception ex)
+            {
+                logger.LogError("Failed to generate aad token. " + ex.Message);
+            }
+            InitializeOpenSourceClient(opsAuthToken);
+            this.logger = logger;
+        }
+
+        private readonly string opsAuthToken;
+        private static HttpClient client;
+        private readonly ILogger<GitHubNameResolver> logger;
+
+        /// <summary>
+        /// Get the user principal name from github. User principal name is in format of ms email.
+        /// </summary>
+        /// <param name="githubUserName">github user name</param>
+        /// <returns>Aad user principal name</returns>
+        public string GetUserPrincipalNameFromGithub(string githubUserName)
+        {
+            try
+            {
+                var reponseJsonString = client.GetStringAsync($"https://repos.opensource.microsoft.com/api/people/links/github/{githubUserName}").Result;
+                dynamic contentJson = JsonConvert.DeserializeObject(reponseJsonString);
+                return contentJson.aad.userPrincipalName;
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex.Message);
+                return null;
+            }
+        }
+
+        private static void InitializeOpenSourceClient(string opsAuthToken)
```

Comment:
Why not just do all these steps directly in the constructor?

Reply:
It is my preference to have less codes in constructor. Since the initialize client method logic is not that complicated, I think I can put all codes back to the constructor.